lib/web/scripts/install.go

```diff
@@ -0,0 +1,167 @@
+	// updates. The script lives in the 'teleport-ent-update' package and was our original attempt at automatic updates.
+	// See RFD-109 for more details: https://github.com/gravitational/teleport/blob/master/rfd/0109-cloud-agent-upgrades.md
+	PackageManagerAutoupdate
+	// UpdaterBinaryAutoupdate means the installed Teleport should update via the teleport-update binary.
+	// This update style does not depend on any package manager (although it has a system dependency to wake up the
+	// updater).
+	// See RFD-184 for more details: https://github.com/gravitational/teleport/blob/master/rfd/0184-agent-auto-updates.md
+	UpdaterBinaryAutoupdate
+)
+
+// InstallScriptOptions contains the Teleport installation options used to generate installation scripts.
+type InstallScriptOptions struct {
+	AutoupdateStyle AutoupdateStyle
+	// TeleportVersion that should be installed. Without the leading "v".
+	TeleportVersion string
+	// CDNBaseURL is the URL of the CDN hosting teleport tarballs.
+	// If left empty, the 'teleport-update' installer will pick the one to use.
+	// For example: "https://cdn.example.com"
+	CDNBaseURL string
+	// ProxyAddr is the address of the Teleport Proxy service that will be used
+	// by the updater to fetch the desired version. Teleport Addrs are
+	// 'hostname:port' (no scheme nor path).
+	ProxyAddr string
+	// TeleportFlavor is the name of the Teleport artifact fetched from the CDN.
+	// Common values are "teleport" and "teleport-ent".
+	TeleportFlavor string
+	// FIPS represents if the installed Teleport version should use Teleport
+	// binaries built for FIPS compliance.
+	FIPS bool
+}
+
+// Check validates that the minimal options are set.
+func (o *InstallScriptOptions) Check() error {
+	switch o.AutoupdateStyle {
+	case NoAutoupdate, PackageManagerAutoupdate:
+		return nil
+	case UpdaterBinaryAutoupdate:
+		// We'll do the checks later.
+	default:
+		return trace.BadParameter("unsupported autoupdate style: %v", o.AutoupdateStyle)
+	}
+	if o.ProxyAddr == "" {
+		return trace.BadParameter("Proxy address is required")
+	}
+
+	if o.TeleportVersion == "" {
+		return trace.BadParameter("Teleport version is required")
+	}
+
+	if o.TeleportFlavor == "" {
+		return trace.BadParameter("Teleport flavor is required")
+	}
+
+	if o.CDNBaseURL != "" {
+		url, err := url.Parse(o.CDNBaseURL)
+		if err != nil {
+			return trace.Wrap(err, "failed to parse CDN base URL")
+		}
+		if url.Scheme != "https" {
+			return trace.BadParameter("CDNBaseURL's scheme must be 'https://'")
+		}
+	}
+	return nil
+}
+
+// oneOffParams returns the oneoff.OneOffScriptParams that will install Teleport
+// using the oneoff.sh script to download and execute 'teleport-update'.
+func (o *InstallScriptOptions) oneOffParams() (params oneoff.OneOffScriptParams) {
+	// We add the leading v if it's not here
+	version := o.TeleportVersion
+	if o.TeleportVersion[0] != 'v' {
+		version = "v" + o.TeleportVersion
+	}
```

Comment:
Why not make `TeleportVersion` a `semver.Version` instead?

Reply:
I'll address this die quest in a separate PR